app/backend/approaches/chatreadretrieveread.py

```diff
@@ -99,16 +99,21 @@ def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]) -> A
         else:
             prompt = prompt_override.format(sources=content, chat_history=self.get_chat_history_as_text(history), follow_up_questions_prompt=follow_up_questions_prompt)
 
+        messages = self.get_messages_from_prompt(prompt)
+
         # STEP 3: Generate a contextual and content specific answer using the search results and chat history
-        completion = openai.Completion.create(
-            engine=self.chatgpt_deployment, 
-            prompt=prompt, 
+        chatCompletion = openai.ChatCompletion.create(
+            deployment_id=self.chatgpt_deployment,
+            model="gpt-3.5-turbo",
```

Comment:
Developers may override the model (like to use gpt-4), so it'd be nice to not hardcode it. It could be an additional environment variable, like we do for the other sample-

https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/app.py#L42

Reply:
agreed. I have added a new environment variable to taken in model name and pass to downstream components. However we need to update the value from provisioning. I will updat them in my next PR